File: Programmers/Lv.2/올바른 괄호/12909.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '(') {
                stack.push(c);
            } else {
                if (stack.isEmpty()) {
                    stack.push(c);
                    break;
                }
                stack.pop();
            }
        }
        
        answer = stack.isEmpty();
        
        return answer;
    }
}

File: Programmers/Lv.2/올바른 괄호/12909.js
function solution(s){
    let answer = true;
    
    const stack = [];
    
    for (const str of s) {
        if (str === '(') {
            stack.push(str);
        
        } else {
            if (stack.length === 0) {
                stack.push(str);
                break;
            }
            stack.pop();
        }
    }
    
    answer = (stack.length === 0);

    return answer;
}
